### PR TITLE
Fix benchmark read/write key tracker for keys in child storages.

### PR DESCRIPTION
**Pull request by thiolliere**
This fix the issues related to count read and write in child tries:
* every write into a child trie will now be counted
* the same key read from different trie won't be counted as repeat

TODO:
* [x] implement some tests
* [x] maybe better log: are current log used in some script or can I just break them ? @shawntabrizi 
* [ ] maybe think if child trie read/writes should be priced differently (should be done in another PR)

### CODE REVIEW

**Comment by shawntabrizi** [comment]
@thiolliere Logs are not used in any script, so feel free to break them

**Comment by thiolliere** [comment]
I don't think this PR needs audit, what it does is simply:
* add one parameter to backend trait in primitive
* change bench backend (used only in benchmarks)

File: client/db/src/bench.rs [hunk 1]
```diff
@@ -200,40 +212,71 @@ impl<B: BlockT> BenchmarkingState<B> {
...
 
-	fn add_write_key(&self, key: &[u8]) {
-		log::trace!(target: "benchmark", "Write: {}", HexDisplay::from(&key));
-
-		let mut key_tracker = self.key_tracker.borrow_mut();
+	// Childtrie is identified by its storage key (i.e. `ChildInfo::storage_key`)
+	fn add_write_key(&self, childtrie: Option<&[u8]>, key: &[u8]) {
 		let mut read_write_tracker = self.read_write_tracker.borrow_mut();
+		let mut child_key_tracker = self.child_key_tracker.borrow_mut();
+		let mut main_key_tracker = self.main_key_tracker.borrow_mut();
```
Comment (shawntabrizi) [inline client/db/src/bench.rs:239]: Would it be better to only borrow this value in appropriate branch the if statement below?

**Review by shawntabrizi — APPROVED** [review]
Thanks for the fix!

File: client/db/src/bench.rs (hunk 1 above)
Comment (thiolliere) [reply client/db/src/bench.rs:239]: actually if we borrow inside the branch then the borrow doesn't live long enough

**Comment by gavofyork** [comment]
bot merge

**Pull request merged**